controller/controller.go

```diff
@@ -0,0 +1,63 @@
+package controller
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/szuecs/kube-static-egress-controller/provider"
+)
+
+// EgressController is the controller for creating Egress configuration via a
+// provider.
+type EgressController struct {
+	interval     time.Duration
+	configsChan  <-chan provider.EgressConfig
+	configsCache map[provider.Resource]map[string]struct{}
+	provider     provider.Provider
+}
+
+// NewEgressController initializes a new EgressController.
+func NewEgressController(prov provider.Provider, configsChan <-chan provider.EgressConfig, interval time.Duration) *EgressController {
+	return &EgressController{
+		interval:     interval,
+		configsChan:  configsChan,
+		provider:     prov,
+		configsCache: make(map[provider.Resource]map[string]struct{}),
+	}
+}
+
+// Run runs the EgressController main loop.
+func (c *EgressController) Run(ctx context.Context) {
+	log.Info("Running controller")
+	interval := c.interval
```

Comment:
why do you need to store interval locally here and in all case branches?

Reply:
This is done to extend the interval in case we just reconciled the stack because of a configmap event or so. I.e. if the interval is 5 min and we get an event and update the stack after 4m30s then we don't need to update the stack only after another 30s.

Reply:
Ah optimizations, hmm, I guess we don’t really need this, but if you already have it.